### PR TITLE
Add tests setup

### PR DESCRIPTION
**Pull request by saleem-hadad**
Salam @themsaid,
Thanks for sharing this package, I've added the initial tests setup.

### CODE REVIEW

**Comment by saleem-hadad** [comment]
This is a snapshot from my package's UI 🤓 I'll release it soon
blog

**Comment by saleem-hadad** [comment]
and what do you think if we combine the two packages since we each one of us implemented a lot of features 💃🤠

**Comment by 1e4** [comment]
You seem to have added the tests under `require` and not `require-dev` in `composer.json`. Production setups do not need these extra libs, I would recommend sending a patch to change this. Other than that it looks like a good start.

File: composer.json [hunk 1]
```diff
@@ -11,7 +11,17 @@
     ],
     "require": {
         "php": "^7.1",
-        "laravel/framework": "~5.7"
+        "illuminate/support": "~5.7",
```
Comment (toniperic) [inline composer.json]: Don't you get `illuminate/support` as subtree split of Support component within `laravel/framework` (which is already required)?

File: composer.json [hunk 2]
```diff
@@ -11,7 +11,17 @@
...
+        "laravel/framework": "~5.7",
+        "mockery/mockery": "^1.2",
+        "orchestra/database": "^3.6",
+        "orchestra/testbench": "^3.7",
+        "phpunit/phpunit": "^7.4"
```
Comment (toniperic) [inline composer.json]: Are some of these perhaps dev dependencies?

File: composer.json (hunk 1 above)
Comment (saleem-hadad) [reply composer.json]: Yes, you're right! 👍

File: composer.json (hunk 2 above)
Comment (saleem-hadad) [reply composer.json]: Yes they should be in the dev, sorry didn't notice 👍

**Comment by saleem-hadad** [comment]
I've opened my package: https://github.com/saleem-hadad/blogged

**Comment by 1e4** [comment]
> I've opened my package: https://github.com/saleem-hadad/blogged

Can you keep this issue as about the tests and not self advertisement, it clutters and detracts the conversation?

**Comment by saleem-hadad** [comment]
Okay. Sorry.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+language: php
+
+sudo: true
+
+php:
+- 7.2
+
+services:
+  - sqlite
```
Comment (lucasmichot) [inline .travis.yml]: The `sqlite` service is not required to run tests

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+before_script:
+  - composer install
+  - travis_retry composer self-update
+  - travis_retry composer update --no-interaction --prefer-dist
```
Comment (lucasmichot) [inline .travis.yml]: You might also want to add the following flag: `--no-suggest`

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+php:
+- 7.2
+
+services:
+  - sqlite
+
+before_script:
+  - composer install
+  - travis_retry composer self-update
```
Comment (lucasmichot) [inline .travis.yml]: This is not required, Travis already update `composer` once in a while

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  - travis_retry composer update --no-interaction --prefer-dist
+  - composer show laravel/framework
```
Comment (lucasmichot) [inline .travis.yml]: I don't think that `composer show laravel/framework` is necessary

File: composer.json [hunk 7]
```diff
@@ -11,7 +11,18 @@
     ],
     "require": {
         "php": "^7.1",
-        "laravel/framework": "~5.7"
+        "illuminate/support": "~5.7"
```
Comment (lucasmichot) [inline composer.json]: Why?

File: tests/Integration/WinkPostTest.php [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+namespace Wink\Tests\Integration;
+
+use Wink\WinkPage;
+use Wink\WinkPost;
+use Wink\Tests\TestCase;
+
+class WinkPostTest extends TestCase
+{
+    /** @test */
+    public function a_guest_may_not_hit_any_post_endpoint()
```
Comment (lucasmichot) [inline tests/Integration/WinkPostTest.php]: I think the Laravel naming convention would be `test_a_guest_may_not_hit_any_post_endpoint()` without the `/** @test */` docblock

File: tests/Integration/WinkPostTest.php [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+     *
+     * @return array
+     */
+    protected function winkPostData($data)
+    {
+        return array_merge([
+            'author_id'    => 1,
+            'title'        => 'my amazing post ;p',
+            'slug'         => 'post-slug',
+            'publish_date' => date(now()),
```
Comment (lucasmichot) [inline tests/Integration/WinkPostTest.php]: or `now()->toDateTimeString(),`

File: tests/TestCase.php [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+
+class TestCase extends \Orchestra\Testbench\TestCase
+{
+    use DatabaseTransactions;
+
+    /**
+     * @return void
+     */
+    public function setUp()
```
Comment (lucasmichot) [inline tests/TestCase.php]: This should be `protected`

File: tests/Unit/WinkPostTest.php [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+        $this->expectException('\Illuminate\Database\QueryException');
+        $winkPost = factory(WinkPost::class)->create(['id' => 'uuid']);
+    }
+
+    /** @test */
+    public function it_has_title()
+    {
+        $winkPost = factory(WinkPost::class)->create(['title' => 'custom title']);
+
+        $this->assertEquals('custom title', $winkPost->title);
```
Comment (lucasmichot) [inline tests/Unit/WinkPostTest.php]: Why not `$this->assertSame` then?

File: tests/Unit/WinkPostTest.php [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+            factory(WinkPost::class)->create(['featured_image' => 'custom_featured_image'])->featured_image
+        );
+    }
+
+    /** @test */
+    public function it_has_author()
+    {
+        $winkPost = factory(WinkPost::class)->create();
+
+        $this->assertInstanceOf(\Wink\WinkAuthor::class, $winkPost->author);
```
Comment (lucasmichot) [inline tests/Unit/WinkPostTest.php]: Maybe you can import `WinkAuthor` ?

**Comment by lucasmichot** [comment]
See #109

**Comment by themsaid** [comment]
sorry :)

**Comment by saleem-hadad** [comment]
no problem

**Pull request closed without merging**